### PR TITLE
feat: Cherry Tracks DataFixer

### PR DESCRIPTION
**Pull request by IThundxr**

### CODE REVIEW

File: common/src/main/java/com/railwayteam/railways/base/datafixers/CompatCherryTrackFix.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import java.util.Optional;
+
+/**
+ * Also see snr$updateCherryCompatTracks inside {@link MixinTrackMaterial}
+ */
+public class CompatCherryTrackFix extends DataFix {
+    /*
+     * List of modded cherry tracks that need to be fixed
+     */
+    public static List<String> standardCherryOld = List.of(
```
Comment (techno-sam) [inline common/src/main/java/com/railwayteam/railways/base/datafixers/CompatCherryTrackFix.java]: these could probably be made final to help the jvm/compiler optimize stuff

File: common/src/generated/resources/assets/railways/lang/en_us.json [hunk 2]
```diff
@@ -551,167 +533,6 @@
 	"railways.ponder.train_switch.text_6": "Switch direction can also be set by a redstone signal, and can be locked by powering the bottom",
 	"railways.ponder.train_switch.text_7": "Projectiles also cycle switches on impact",
 
-	"create.railways.semaphore.search_mode": "Signal Search Direction",
```
Comment (techno-sam) [inline common/src/generated/resources/assets/railways/lang/en_us.json:554]: why is all this getting removed?

File: common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
 @Mixin(value = TrackMaterial.class, remap = false)
 public class MixinTrackMaterial {
+    /**
+     * DataFixer for Modded Compat Cherry Tracks.
+     * If it finds a modded cherry track material it'll replace it {@link CRTrackMaterials#CHERRY }
+     * Which is from 1.20 Vanilla.
+     */
+    @Inject(method = "deserialize", at = @At("HEAD"), cancellable = true)
+    private static void snr$updateCherryCompatTracks(String serializedName, CallbackInfoReturnable<TrackMaterial> cir) {
+        List<String> cherryList = List.of("biomesoplenty:cherry", "byg:cherry", "blue_skies:cherry");
```
Comment (techno-sam) [inline common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java]: why aren't we handling narrow and wide variants?

File: gradle.properties [hunk 4]
```diff
@@ -101,8 +101,8 @@ enable_byg = false
 
 # Biomes O' Plenty
 # https://www.curseforge.com/minecraft/mc-mods/biomes-o-plenty/files
-bop_version = 4617259
-enable_bop = false
+bop_version = 4764804
+enable_bop = true
```
Comment (techno-sam) [inline gradle.properties]: make sure to turn this off after you are done testing

**Review by techno-sam — CHANGES_REQUESTED** [review]
Good! I left a few notes

File: common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java (hunk 3 above)
Comment (IThundxr) [reply common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java]: Track Materials dont have narrow/wide gauges its just a single material all gauges share

File: common/src/generated/resources/assets/railways/lang/en_us.json (hunk 2 above)
Comment (IThundxr) [reply common/src/generated/resources/assets/railways/lang/en_us.json:554]: In one of raz's PRs intellij seemed to insert translations from en_us.json into interface.json, i removed those so they wouldnt duplicate etc

Comment (techno-sam) [reply common/src/generated/resources/assets/railways/lang/en_us.json:554]: I feel like we might be losing translations, but we'll figure that out after this is merged

File: common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java (hunk 3 above)
Comment (techno-sam) [reply common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java]: wrong. gauges are separate materials, mapping from normal to wide is `CRTrackMaterials.WIDE_GAUGE` and ditto for narrow gauge. the material id is `<namespace>:<base_path>_wide` and `<namespace>:<base_path>_narrow`

File: common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java [hunk 5]
```diff
@@ -21,14 +21,18 @@ public class MixinTrackMaterial {
...
     private static void snr$updateCherryCompatTracks(String serializedName, CallbackInfoReturnable<TrackMaterial> cir) {
-        List<String> cherryList = List.of("biomesoplenty:cherry", "byg:cherry", "blue_skies:cherry");
-
-        if (cherryList.contains(serializedName))
-            cir.setReturnValue(CRTrackMaterials.CHERRY);
+        switch (serializedName) {
+            case "biomesoplenty:cherry", "byg:cherry", "blue_skies:cherry"
+                    -> cir.setReturnValue(CRTrackMaterials.CHERRY);
+            case "biomesoplenty:cherry_wide", "byg:cherry_wide", "blue_skies:cherry_wide"
+                    -> cir.setReturnValue(CRTrackMaterials.WIDE_GAUGE.get(CRTrackMaterials.CHERRY));
```
Comment (techno-sam) [inline common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java]: this works, but it's better to use the helper functions `CRTrackMaterials.getWide(<base>)` and `CRTrackMaterials.getNarrow(<base>)`

**Review by techno-sam — CHANGES_REQUESTED** [review]
almost there!

File: common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java (hunk 5 above)
Comment (IThundxr) [reply common/src/main/java/com/railwayteam/railways/mixin/MixinTrackMaterial.java]: Fixed!

**Review by techno-sam — APPROVED** [review]
Looks good!

**Pull request merged**